sympy/solvers/diophantine.py

```diff
@@ -1004,7 +1004,7 @@ def _diop_quadratic(var, coeff, t):
                 for z0 in range(0, abs(_c)):
                     # Check if the coefficients of y and x obtained are integers or not
                     if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
-                            divisible(e*sqc**g*z0**2 + E*z0 + e*sqc*F, _c)):
+                            divisible(e*sqc*g*z0**2 + E*z0 + e*sqc*F, _c)):
```

Comment:
This was a good catch. If it hasn't been fixed in master and no tests failed, the following test should be added (and I confirmed that the solution satisfies, in general, the original equation):
```python
assert diophantine(8*x**2 - 24*x*y + 18*y**2 + 5*x + 7*y + 16) == {
    (-174*t**2 + 41*t - 4, -116*t**2 + 37*t - 4),
    (-174*t**2 + 17*t - 2, -116*t**2 + 21*t - 2)}
```
That test fails in 1.5.

Reply:
cf https://github.com/sympy/sympy/issues/18156

Reply:
That typo has been already fixed in [#18130](https://github.com/sympy/sympy/pull/18130#issue-356831749). My sympy version is '1.6.dev'. That test is successful in that version.